src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -79,4 +95,485 @@ protected JacobianMatrix createJacobianMatrix(EquationSystem equationSystem, Vol
         equationSystem.updateEquations(x);
         return new JacobianMatrix(equationSystem, matrixFactory);
     }
+
+    protected void fillRhsSensitivityVariable(LfNetwork lfNetwork, EquationSystem equationSystem, List<SensitivityFactorGroup> factorGroups, Matrix rhs) {
+        for (SensitivityFactorGroup factorGroup : factorGroups) {
+            factorGroup.fillRhs(lfNetwork, equationSystem, rhs);
+        }
+    }
+
+    protected List<ParticipatingElement> getParticipatingElements(LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters, OpenLoadFlowParameters openLoadFlowParameters, Function<ParticipatingElement, Boolean> filter) {
+        ActivePowerDistribution.Step step = ActivePowerDistribution.getStep(loadFlowParameters.getBalanceType(), openLoadFlowParameters.isLoadPowerFactorConstant());
+        List<ParticipatingElement> participatingElements =  step.getParticipatingElements(lfNetwork).stream().filter(filter::apply).collect(Collectors.toList());
+        ParticipatingElement.normalizeParticipationFactors(participatingElements, "bus");
+        return participatingElements;
+    }
+
+    protected List<ParticipatingElement> getParticipatingElements(LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters, OpenLoadFlowParameters openLoadFlowParameters) {
+        return getParticipatingElements(lfNetwork, loadFlowParameters, openLoadFlowParameters, element -> true);
+    }
+
+    static class LfSensitivityFactor<T extends EquationTerm> {
```

Comment:
Indeed this is a bit weird but it works. @floriand-e2r is it the good way to handle that?

Reply:
This is not ideal, as we have a lot of "raw type" warning from Sonar. Let's see that for a later refactoring.